### PR TITLE
Add cookbook entry for Gaussian Naive Bayes example

### PR DESCRIPTION
**Pull request by tbskm**
Added a cookbook entry for Gaussian Naive Bayes. Also noticed that the sample directory for `knn.rst` in the cookbook `README.md` wasn't quite right, so I updated it. If this should be changed in its own PR, please let me know.

@karlnapf Could you clarify what you meant by "No need to include the translates files." on the last PR? I'm not totally sure what the translate files would be in this case. Thanks!

### CODE REVIEW

File: doc/cookbook/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ To add an entry, only two files are needed (plus a test):
...
+ * A Sphinx markdown file with matching filename and directory, e.g. ```shogun/doc/cookbook/source/examples/classifier/knn.rst```. This file contains a description of the API example and references to code snippets. The point is to *not* show the full file listing but only a subset. The file should furthermore contain basic math in the form of LaTeX, important references (wikipedia, scientific paper references using BibTeX, etc), and links to the involved Shogun class documentation, github issues, etc. We use Sphinx tags to make this easy, see for example [knn.rst](https://github.com/shogun-toolbox/shogun/blob/develop/doc/cookbook/source/examples/classifier/knn.rst).
```
Comment (karlnapf) [inline doc/cookbook/README.md]: thanks for the correction!

**Comment by karlnapf** [comment]
"No need to include the translates files." -- in your previous PR there was all the python, R, octave files etc. Which is not necessary, but they are not in this PR so dont worry

**Comment by karlnapf** [comment]
Your example is not an executable listings. Make sure to run `make test` with having `-DENABLE_TESTING=ON` in your cmake. 
You will see that the generated cpp doesnt compile

File: doc/cookbook/source/examples/classifier/gaussian_naive_bayes.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Imagine we have files with training and test data. We create CDenseFeatures (here 64 bit floats aka RealFeatures) and :sgclass:`CMulticlassLabels` as
+
+.. sgexample:: gaussiannaivebayes.sg:create_features
+
+We create an instance of the :sgclass:`CGaussianNaiveBayes` classifier, passing it training data and the label.
+
+.. sgexample:: gaussiannaivebayes.sg:create_instance
+
+Then we run the train Gaussian Naive Bayes algorithm, apply it to the test data, and print the prediction.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/classifier/gaussian_naive_bayes.rst]: see the updates I made to knn.rst, and also to the Readme. They are minor, but could you please mimic them?
Not printing here. But rather say what comes out of the precition method

File: examples/meta/src/classifier/gaussian_naive_bayes.sg [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+#![create_features]
+RealFeatures features_train(f_feats_train)
+RealFeatures features_test(f_feats_test)
+MulticlassLabels labels_train(f_labels_train)
+MulticlassLabels labels_test(f_labels_test)
+#![create_features]
+
+
+#![create_instance]
+GaussianNaiveBayes gnb(features_train)
```
Comment (karlnapf) [inline examples/meta/src/classifier/gaussian_naive_bayes.sg]: there is no such constructor in the class, that is why is doesnt compile

**Comment by karlnapf** [comment]
After you have run `make test` the cpp example will have produced a `.dat` file. Can you add this one to the testsuite in data and update the data version of this PR? Then we have an integration test for gaussian naive bayes

**Comment by karlnapf** [comment]
Finally, remove all existing gaussian naive bayes examples in the undocumented examples folder

**Comment by karlnapf** [comment]
Just merged the linear ridge regression one. Seek inspiration from the patch #3064

**Comment by tbskm** [comment]
Whoops, didn't mean to leave the labels out of the constructor, my mistake. Thanks for being patient, I'll get to work on making these changes.

**Comment by tbskm** [comment]
Closing this to submit a PR with the above fixes.

**Pull request closed without merging**